### DISCUSSION TITLE
RTL grid

### DISCUSSION THREAD

**Discussion by avidseeker** [ImageMagick/ImageMagick#7692] [General]
Issue is similar to the PDF case: https://tex.stackexchange.com/questions/727552/

Consider pages of a RTL book: `001.png, 002.png, 003.png, ...`

Since the book is RTL, the first page starts on the right not the left. Commands like

```
$ montage *.png -tile 2x2 -geometry +2+2 out.png
```

result in 
```
---------
| 1 | 2 |
| 3 | 4 |
---------
```

Instead of 

```
---------
| 2 | 1 |
| 4 | 3 |
---------
```

A possible workaround might be a script that re-orders the pages.

```
$ montage $(echo *.png | rtl_fix.sh)
```

**Comment by snibgo** [comment]
Yes, a script that renames the input files, or creates symbolic links with appropriate names, is a reasonable solution.

An alternative is to `-flop` each input to `montage`, and then `-flop` the outputs:
```
magick montage *.png -flop miff:- | magick - -flop x.png
```
This solution is heavy on CPU usage.

**Comment by snibgo** [comment]
Another method, that isn't heavy on the CPU, is:
```
magick montage *.png -swap 0,1 -swap 2,3 -swap 4,5 x.png
```
Repeat the swap as many times as required.
